tour_model_eval/trips_in_bins_and_clusters.ipynb

```diff
@@ -42,293 +49,789 @@
+    "radius = 300"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "prospective-conducting",
+   "id": "dependent-machinery",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "trips = pipeline.read_data(uuid=user)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "steady-consultation",
+   "metadata": {},
+   "source": [
+    "Note: Run trips on all data and trips in bins/clusters above cutoff point separately "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "exotic-timing",
    "metadata": {},
+   "source": [
+    "## Visualizing trips from bins"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "recovered-peoples",
+   "metadata": {},
+   "source": [
+    "### Trips from all bins"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "ultimate-bottom",
+   "metadata": {
+    "scrolled": true
+   },
+   "outputs": [],
+   "source": [
+    "sim = similarity.similarity(trips, radius)\n",
+    "sim.bin_data()"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "amino-particular",
+   "metadata": {
+    "scrolled": true
+   },
    "outputs": [],
    "source": [
-    "import emission.analysis.modelling.tour_model.cluster_pipeline as eamtc"
+    "logging.debug('There are %s bins before filtering' % len(sim.bins))"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "id": "finnish-istanbul",
+   "id": "reasonable-sharp",
+   "metadata": {
+    "scrolled": false
+   },
+   "outputs": [],
+   "source": [
+    "#Create colormap for all bins\n",
+    "cmp_all_bins = cm.linear.Set1_07.to_step(len(sim.bins),index=[i for i in range (len(sim.bins))])\n",
+    "cmp_all_bins"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "incorporated-plymouth",
    "metadata": {},
    "outputs": [],
    "source": [
-    "# we should experiment with different values here\n",
-    "radius = 300"
+    "# Plot all the bin trips on the map, use different color for different bins.\n",
+    "m = folium.Map(location=[trips[0].data.start_loc[\"coordinates\"][1], trips[0].data.start_loc[\"coordinates\"][0]],zoom_start=12, max_zoom= 30, control_scale=True)\n",
+    "for t in range (len(sim.bins)):    \n",
+    "    for i in range (len(sim.bins[t])):\n",
+    "        layer = folium.PolyLine([[trips[sim.bins[t][i]].data.start_loc[\"coordinates\"][1],trips[sim.bins[t][i]].data.start_loc[\"coordinates\"][0]],\n",
+    "                                [trips[sim.bins[t][i]].data.end_loc[\"coordinates\"][1],trips[sim.bins[t][i]].data.end_loc[\"coordinates\"][0]]],weight=2, \n",
+    "                                color=cmp_all_bins(t))\n",
+    "        layer.add_to(m)\n",
+    "m.add_child(cmp_all_bins)     \n",
+    "m"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "better-continent",
+   "metadata": {},
+   "source": [
+    "### Trips from bins above the cutoff only"
```

Comment:
As I said, please move this to a separate notebook. In general, it is good to follow the [Golden Rule](https://en.wikipedia.org/wiki/Golden_Rule) while writing code - think about the level of clarity and commenting that you would have liked to see in Naomi's code, and introduce it into your code 🙏

Think about the next person who will work on this code and have questions like "what is this supposed to be doing?"